### DISCUSSION TITLE
How to enable verbose debugging in VS2022 for Avalonia

### DISCUSSION THREAD

**Discussion by bsdRuf** [AvaloniaUI/Avalonia#13836] [Q&A] [answered]
Is there a way to enable more verbose error messages?

The reason why I need to ask for it is that ma MainWindow.axaml isnt loading anymore and throws:

"Invalid Markup - Check the Error List for more information."

Error List:
"Error		Object reference not set to an instance of an object.	%userpath%\MainWindow.axaml		[ProjectFolder]"

There is no Line or Hint what I forgot to set and I am very sure the last changes didnt affect the preview. I loaded the old version from my local Gitlab and even the version before the changes happend now cause the preview error.

I cant reproduce it because my previous Version also gets the error but I was reworking and styling the Windows, so I am sure the previewer was working in that session.

Or any ideas how I can find out what my mistake was?

Thanks in advance

**Comment by thevortexcloud** [comment, marked as answer]
If this is a runtime error you can often get Visual Studio to stop at the exception if you enable that specific exception type in the exception settings. It often will show you the exact line in the XAML.

If it's a compile time error I am not sure.

**Reply by bsdRuf** [reply]
Thank you a lot.
Found the line that caused the error. I was selecting sth from my DB for the Combobox.

**Reply by timunie** [reply]
> [!TIP]
> use `if(Design.IsDesignTime) ...` to skip I/O operations.

**Comment by maxkatz6** [comment]
In 11.1 (nightly builds) it's:
```csproj
<AvaloniaXamlReportImportance>High</AvaloniaXamlReportImportance>
<AvaloniaXamlVerboseExceptions>true</AvaloniaXamlVerboseExceptions>
```
